src/Analyzers/CSharp/CodeFixes/UsePrimaryConstructor/CSharpUsePrimaryConstructorCodeFixProvider.cs

```diff
@@ -129,26 +125,16 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         await MoveBaseConstructorArgumentsAsync().ConfigureAwait(false);
 
         // Then take all the assignments in the constructor, and place them directly on the field/property initializers.
-        if (constructorDeclaration.ExpressionBody is not null)
-        {
-            // Validated by analyzer.
-            await ProcessAssignmentAsync((AssignmentExpressionSyntax)constructorDeclaration.ExpressionBody.Expression, expressionStatement: null).ConfigureAwait(false);
-        }
-        else
-        {
-            Contract.ThrowIfNull(constructorDeclaration.Body);
-            foreach (var statement in constructorDeclaration.Body.Statements)
-            {
-                // Validated by analyzer.
-                var expressionStatement = (ExpressionStatementSyntax)statement;
-                await ProcessAssignmentAsync((AssignmentExpressionSyntax)expressionStatement.Expression, expressionStatement).ConfigureAwait(false);
-            }
-        }
+        await ProcessConstructorAssignmentsAsync().ConfigureAwait(false);
 
         // Then remove the constructor itself.
         var constructorDocumentEditor = await solutionEditor.GetDocumentEditorAsync(document.Id, cancellationToken).ConfigureAwait(false);
         constructorDocumentEditor.RemoveNode(constructorDeclaration);
 
+        // When moving the parameter list from the constructor to the type, we will no longer have nested types or
+        // member constants in scope.  So rewrite references to them if that's the case.
+        var updatedParameterList = GenerateFinalParameterList();
```

Comment:
extracted to helper funciton, and pulled out of callback lambda below.  it doesn't use any of hte updated callback state.